mrmustard/lab/gates.py

```diff
@@ -430,93 +430,103 @@ def _validate_modes(self, modes):
+        if unitary is None:
+            unitary = math.random_unitary(num_modes)
         super().__init__(
-            orthogonal=orthogonal,
-            orthogonal_trainable=orthogonal_trainable,
+            unitary=unitary,
+            unitary_trainable=unitary_trainable,
         )
         self._modes = modes or list(range(num_modes))
         self.is_gaussian = True
 
     @property
     def X_matrix(self):
-        return self.orthogonal.value
+        return math.block(
+            [
+                [math.real(self.unitary.value), -math.imag(self.unitary.value)],
+                [math.imag(self.unitary.value), math.real(self.unitary.value)],
+            ]
+        )
 
     def _validate_modes(self, modes):
-        if len(modes) != self.orthogonal.value.shape[-1] // 2:
+        if len(modes) != self.unitary.value.shape[-1]:
             raise ValueError(
-                f"Invalid number of modes: {len(modes)} (should be {self.orthogonal.shape[-1] // 2})"
+                f"Invalid number of modes: {len(modes)} (should be {self.unitary.shape[-1]})"
             )
 
     def __repr__(self):
         modes = self.modes
-        orthogonal = repr(math.asnumpy(self.orthogonal.value)).replace("\n", "")
-        return f"Interferometer(num_modes = {len(modes)}, orthogonal = {orthogonal}){modes}"
+        unitary = repr(math.asnumpy(self.unitary.value)).replace("\n", "")
+        return f"Interferometer(num_modes = {len(modes)}, unitary = {unitary}){modes}"
 
 
 class RealInterferometer(Parametrized, Transformation):
-    r"""N-mode interferometer with a real unitary matrix (or block-diagonal orthogonal matrix).
+    r"""N-mode interferometer with a real unitary matrix (or block-diagonal unitary matrix).
     Does not mix q's and p's.
 
     Args:
-        orthogonal (2d array, optional): a valid orthogonal matrix. For N modes it must have shape `(N,N)`.
-            If set to `None` a random orthogonal matrix is used.
-        orthogonal_trainable (bool): whether orthogonal is a trainable variable
+        unitary (2d array, optional): a valid unitary matrix. For N modes it must have shape `(N,N)`.
+            If set to `None` a random unitary matrix is used.
+        unitary_trainable (bool): whether unitary is a trainable variable
     """
 
     def __init__(
         self,
         num_modes: int,
-        orthogonal: Optional[Tensor] = None,
-        orthogonal_trainable: bool = False,
+        unitary: Optional[Tensor] = None,
+        unitary_trainable: bool = False,
+        modes: Optional[List[int]] = None,
     ):
-        if orthogonal is None:
-            orthogonal = math.random_orthogonal(num_modes)
-        super().__init__(orthogonal=orthogonal, orthogonal_trainable=orthogonal_trainable)
-        self._modes = list(range(num_modes))
+        if modes is not None and (
+            num_modes != len(modes) or any(mode >= num_modes for mode in modes)
```

Comment:
`mode` can be larger than `num_modes` (what if you want to put a 2-mode inteferometer on modes 30 and 31)?